### ISSUE TITLE
NFT tutorial uses old version of openzeppelin that doesn't compile

### ISSUE THREAD

**Issue by Hamza-Shahzad** [ethereum/ethereum-org-website#3691]
**Describe the bug**

Hello, I was following the [NFT tutorial](https://ethereum.org/en/developers/tutorials/how-to-write-and-deploy-an-nft/), when I faced an issue at [Step 16](https://ethereum.org/en/developers/tutorials/how-to-write-and-deploy-an-nft/#deploy-contract) when the code wouldn't run, here's a look at the error:

![image](https://user-images.githubusercontent.com/13493539/130304888-c240bece-4732-48e2-8ff4-cae780330e72.png)

Looks like the issue is having a different version vs the openzeppelin code which is at 0.8.0. If you simply change the version up top along with the hardhat.config.js file, you'll run into another issue which is that [openzeppelin has taken away the _setTokenURI function](https://forum.openzeppelin.com/t/function-settokenuri-in-erc721-is-gone-with-pragma-0-8-0/5978/3).

Since it's a beginner tutorial I'm not sure if I was supposed to just try to force using the older openzeppelin contracts, but the simple fix I found was to change to the 0.8.0 version, and change [Step 10's code to the following:](https://ethereum.org/en/developers/tutorials/how-to-write-and-deploy-an-nft/#write-contract)

`function mintNFT(address recipient, string memory uri)
        public onlyOwner
        returns (uint256)
    {
        _tokenIds.increment();

        uint256 newItemId = _tokenIds.current();
        _mint(recipient, newItemId);
        uri = tokenURI(newItemId,);

        return newItemId;
    }`

This worked for me, as the contract ultimately deployed. If this looks like a reasonable solution I wouldn't mind submitting the PR, thanks!

**Comment by minimalsm** [comment]
@Hamza-Shahzad thanks. Yeah I tested it out and ended up forcing it through but it probably makes more sense to take your approach. I've assigned you if you fancy slinging at a PR at this 😃

**Comment by minimalsm** [comment]
Solved, thanks!

**Comment by Hamza-Shahzad** [comment]
Hello, sorry about this but I believe I jumped the gun in the solution. tokenURI returns a json file, it doesn't really set anything. I had submitted the PR before I finished the tutorial, and if you use the code I pushed you won't see your NFT published. 

Doing a little more diligence it looks like OpenZeppelin has actually kept the _setTokenURI function, but within the ERC721URIStorage contract instead of within the regular ERC721 contract we were using.

This is what a working MyNFT.sol file looks like:

https://gist.github.com/Hamza-Shahzad/258f09ab824cc0b9656f1492d1e7f579

Additonally here's the folks at OpenZeppelin talking about changing their ERC721 tutorial as well:

https://forum.openzeppelin.com/t/function-settokenuri-in-erc721-is-gone-with-pragma-0-8-0/5978/16

And here's the PR of them adjusting their tutorial

https://github.com/OpenZeppelin/openzeppelin-contracts/pull/2630/commits/f85945a6942302ab95b767d8792304224b2c0af7

Apologies for the confusion!
